presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -223,11 +224,22 @@ public PlanWithProperties visitOutput(OutputNode node, Context context)
         {
             PlanWithProperties child = planChild(node, context.withPreferredProperties(PreferredProperties.undistributed()));
 
+            // if the child is already a gathering exchange, don't add another exchange
+            if ((child.getNode() instanceof ExchangeNode) && ((ExchangeNode) child.getNode()).getType() == ExchangeNode.Type.GATHER) {
+                return rebaseAndDeriveProperties(node, child);
+            }
+
             if (!child.getProperties().isSingleNode()) {
                 child = withDerivedProperties(
                         gatheringExchange(idAllocator.getNextId(), REMOTE, child.getNode()),
                         child.getProperties());
             }
+            else if (!child.getProperties().isCoordinatorOnly()) {
+                // Avoid computation on coordinator
+                child = withDerivedProperties(
```

Comment:
What `isOrderSensitive` flag does is that it only limits parallelism in https://github.com/prestodb/presto/pull/8074/commits/4d52a6fc8f1814951b3e46563379b1f9e7c0a99c#diff-e5a018f043de654e40baeb9ce677d4aaR591.
Instead of adding special `orderSensitive` flag I think there are two approaches that fit better:
1. Explicitly set `context.setDriverInstanceCount(1)` in `LocalExecutionPlanner.Visitor#visitOutput`. I think it's the easiest solution.
2. Perform order sensitiveness property analysis as in `AddLocalExchanges`, (see: `AddLocalExchanges.Rewriter#visitOutput`). It might be an overkill though and I think 1) should be enough.

@martint any thoughts?